packages/@expo/metro-config/src/serializer/serializeChunks.ts

```diff
@@ -381,6 +381,9 @@ export class Chunk {
         premodules = plugin({ graph: this.graph, premodules, debugId });
       }
       this.preModules = new Set(premodules);
+
+      // The filename needs to be updated after the plugins run as they can change the file content.
+      outputFile = this.getFilenameForConfig(serializerConfig);
```

Comment:
This doesn't seem exactly right. The `outputFile` is used to generate the `debugId` which is now outdated if the content changes. This is also a pretty expensive function that we should avoid running multiple times.

Reply:
You are right, the `debugId` is based on the content and that is outdated when the plugin changes it. But it's a chicken or the egg type of problem.

We need to change the content to polyfill the `debugId` functionality (to be able to read it in runtime, from the file without any file io) and the `debugId` is based on the content.

Reply:
For the `debugId` to work and uniquely identify the file only on hook can change the content and add the debugId.

If `getFilenameForConfig(serializerConfig)` would run after each hook it would have to be the last hook to add the `debugId` polyfill. 

Or the hook could be slightly changed as purely `debugId` only/flag to enable the polyfill.